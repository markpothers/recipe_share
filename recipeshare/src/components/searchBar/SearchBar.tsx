import { TextInput, View } from "react-native";
import { responsiveFontSize, responsiveHeight, responsiveWidth } from "react-native-responsive-dimensions"; //eslint-disable-line no-unused-vars

import React from "react";
import SearchBarClearButton from "./searchBarClearButton";
import { centralStyles } from "../../centralStyleSheet"; //eslint-disable-line no-unused-vars

type OwnProps = {
	searchTerm: string;
	text: string;
	setSearchTerm: (searchTerm: string) => void;
	onBlur: () => void;
	searchBar: React.RefObject<typeof SearchBar>;
};

export default function SearchBar(props: OwnProps) {
	const { searchTerm, text, setSearchTerm, searchBar, onBlur } = props;

	// console.log(props)
	return (
		<View
			style={{
				justifyContent: "center",
				alignItems: "center",
				marginLeft: 0,
				marginBottom: responsiveHeight(0.5),
			}}
		>
			<View style={centralStyles.formSection}>
				<View
					style={{
						flexDirection: "row",
						marginTop: responsiveHeight(0.5),
						width: responsiveWidth(100),
						justifyContent: "space-between",
						alignItems: "center",
						backgroundColor: "white",
						borderRadius: responsiveWidth(1.5),
						borderWidth: 1,
						borderColor: "#104e01",
						overflow: "hidden",
					}}
				>
					<TextInput
						maxFontSizeMultiplier={2}
						style={{
							minHeight: responsiveHeight(6),
							backgroundColor: "white",
							textAlign: "left",
							textAlignVertical: "center",
							paddingLeft: responsiveWidth(2),
							width: responsiveWidth(90),
							fontSize: responsiveFontSize(2),
						}}
						value={searchTerm}
						placeholder={text}
						keyboardType="default"
						autoCapitalize="none"
						onChangeText={(text) => setSearchTerm(text)}
						ref={searchBar}
						// onFocus={props.onFocus}
						onBlur={onBlur}
						testID={"searchTermInput"}
					/>
					{searchTerm.length > 0 && <SearchBarClearButton displayIcon={true} setSearchTerm={setSearchTerm} />}
				</View>
			</View>
		</View>
	);
}
