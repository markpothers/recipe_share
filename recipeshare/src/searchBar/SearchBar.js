import React from 'react'
import { View, TextInput } from 'react-native'
import { centralStyles } from '../centralStyleSheet' //eslint-disable-line no-unused-vars
import { responsiveWidth, responsiveHeight, responsiveFontSize } from 'react-native-responsive-dimensions'; //eslint-disable-line no-unused-vars
import Icon from 'react-native-vector-icons/MaterialCommunityIcons'
import { TouchableOpacity } from 'react-native-gesture-handler';

export default function SearchBar(props) {

	// console.log(props)
	return (
		<View
			style={[centralStyles.formContainer,
			{
				width: responsiveWidth(100),
				marginLeft: 0,
				marginBottom: responsiveHeight(0.25),
				// borderWidth: 1
			}
			]}>
			<View style={centralStyles.formSection}>
				<View style={[centralStyles.formInputContainer,
				{
					flexDirection: 'column',
					backgroundColor: 'blue',
					flexWrap: 'nowrap',
					// borderWidth: 1,
					alignItems: null
				}
				]}>
					<TextInput
						maxFontSizeMultiplier={2}
						style={[centralStyles.formInput, { position: 'absolute' }]}
						value={props.searchTerm}
						placeholder={props.text}
						keyboardType="default"
						autoCapitalize="none"
						onChangeText={(text) => props.setSearchTerm(text)}
						ref={props.searchBar}
						// onFocus={props.onFocus}
						onBlur={props.onBlur}
					/>
					{props.searchTerm.length > 0 && (
						<TouchableOpacity style={{
							// backgroundColor: 'red',
							left: responsiveWidth(87),
							width: responsiveWidth(10),
							height: '100%',
							justifyContent: 'center',
							alignItems: 'center',
						}}
							onPress={() => props.setSearchTerm("")}
						>
							<Icon name='close' size={responsiveHeight(3.5)} style={{ color: '#505050' }} />
						</TouchableOpacity>
					)}
				</View>

			</View>
		</View>
	)
}
