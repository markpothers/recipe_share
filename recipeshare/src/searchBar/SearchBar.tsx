import React from "react"
import { View, TextInput } from "react-native"
import { centralStyles } from "../centralStyleSheet" //eslint-disable-line no-unused-vars
import { responsiveWidth, responsiveHeight, responsiveFontSize } from "react-native-responsive-dimensions"; //eslint-disable-line no-unused-vars
import SearchBarClearButton from "./SearchBarClearButton"

type Props = {
	searchTerm: string;
	text: string;
	setSearchTerm: (searchTerm: string) => void;
	onBlur: () => void;
	searchBar: React.RefObject<typeof SearchBar>;
}

export default function SearchBar(props: Props) {
	const {searchTerm, text, setSearchTerm, searchBar, onBlur} = props;

	// console.log(props)
	return (
		<View
			style={
				{
					justifyContent: "center",
					alignItems: "center",
					marginLeft: 0,
					marginBottom: responsiveHeight(0.5),
				}
			}>
			<View style={centralStyles.formSection}>
				<View style={
					{
						flexDirection: "row",
						marginTop: responsiveHeight(0.5),
						width: responsiveWidth(100),
						justifyContent: "space-between",
						alignItems: "center",
						backgroundColor: "white",
						borderRadius: responsiveWidth(1.5),
						borderWidth: 1,
						borderColor: "#104e01",
						overflow: "hidden",
					}
				}>
					<TextInput
						maxFontSizeMultiplier={2}
						style={{
							minHeight: responsiveHeight(6),
							backgroundColor: "white",
							textAlign: "left",
							textAlignVertical: "center",
							paddingLeft: responsiveWidth(2),
							width: responsiveWidth(90),
							fontSize: responsiveFontSize(2)
						}}
						value={searchTerm}
						placeholder={text}
						keyboardType="default"
						autoCapitalize="none"
						onChangeText={(text) => setSearchTerm(text)}
						ref={searchBar}
						// onFocus={props.onFocus}
						onBlur={onBlur}
						testID={"searchTermInput"}
					/>
					{searchTerm.length > 0 && (
						<SearchBarClearButton
							displayIcon={true}
							setSearchTerm={setSearchTerm}
						/>
					)}
				</View>

			</View>
		</View>
	)
}
