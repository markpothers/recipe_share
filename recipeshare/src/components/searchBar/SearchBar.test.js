import "react-native";

import { act, create } from "react-test-renderer";

import React from "react";
import SearchBar from "./searchBar";

describe("Search Bar", () => {
	let component;
	let mockSetFunction = jest.fn();
	let mockBlurFunction = jest.fn();

	beforeAll(() => {
		// console.log('runs at the beginning of everything')
	});

	beforeEach(() => {
		// console.log('runs before every test')
		mockSetFunction = jest.fn();
		mockBlurFunction = jest.fn();
		act(() => {
			component = create(
				<SearchBar
					setSearchTerm={mockSetFunction}
					text={"SearchBar placeholder text"}
					searchTerm={""}
					onBlur={mockBlurFunction}
				/>
			);
		});
	});

	afterEach(() => {
		// console.log('runs after each test')
	});

	afterAll(() => {
		// console.log('runs after all tests have completed')
	});

	test("can be rendered (without the delete button) matches its previous image", () => {
		const image = component.toJSON();
		let root = component.root;
		let button = root.findAllByProps({ testID: "deleteSearchTermButton" });
		expect(button.length).toEqual(0);
		expect(image).toMatchSnapshot();
	});

	test("can be rendered (with the delete button) matches its previous image", () => {
		act(() => {
			component.update(
				<SearchBar
					setSearchTerm={mockSetFunction}
					text={"SearchBar placeholder text"}
					searchTerm={"spaghetti"}
					onBlur={mockBlurFunction}
				/>
			);
		});
		const image = component.toJSON();
		let root = component.root;
		let button = root.findByProps({ testID: "deleteSearchTermButton" });
		expect(button).toBeTruthy();
		expect(image).toMatchSnapshot();
	});

	test("typing in the search bar calls the props set function", () => {
		let root = component.root;
		let input = root.findByProps({ testID: "searchTermInput" });
		input.props.onChangeText("spaghetti");
		expect(mockSetFunction).toHaveBeenCalled();
		expect(mockSetFunction).toHaveBeenCalledWith("spaghetti");
	});

	test("renders with text and the delete button clears the clears the searchTerm of text", () => {
		act(() => {
			component.update(
				<SearchBar
					setSearchTerm={mockSetFunction}
					text={"SearchBar placeholder text"}
					searchTerm={"spaghetti"}
					onBlur={mockBlurFunction}
				/>
			);
		});
		let root = component.root;
		let button = root.findByProps({ testID: "deleteSearchTermButton" });
		button.props.onPress();
		expect(mockSetFunction).toHaveBeenCalledWith("");
	});

	test("calls the blur function when blurred", () => {
		let root = component.root;
		let input = root.findByProps({ testID: "searchTermInput" });
		input.props.onBlur();
		expect(mockBlurFunction).toHaveBeenCalled();
	});
});
