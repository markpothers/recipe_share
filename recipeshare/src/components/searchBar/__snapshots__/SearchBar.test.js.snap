// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar can be rendered (with the delete button) matches its previous image 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "marginBottom": 6.67,
      "marginLeft": 0,
    }
  }
>
  <View
    style={
      {
        "borderColor": "red",
        "borderWidth": 0,
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#104e01",
          "borderRadius": 11.25,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 6.67,
          "overflow": "hidden",
          "width": 750,
        }
      }
    >
      <TextInput
        autoCapitalize="none"
        keyboardType="default"
        maxFontSizeMultiplier={2}
        onBlur={[MockFunction]}
        onChangeText={[Function]}
        placeholder="SearchBar placeholder text"
        style={
          {
            "backgroundColor": "white",
            "fontSize": 30.595932917809698,
            "minHeight": 80.03999999999999,
            "paddingLeft": 15,
            "textAlign": "left",
            "textAlignVertical": "center",
            "width": 675,
          }
        }
        testID="searchTermInput"
        value="spaghetti"
      />
      <View
        accessibilityLabel="clear search text"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "flex-end",
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
        testID="deleteSearchTermButton"
      >
        <Icon
          name="close"
          size={46.690000000000005}
          style={
            {
              "color": "#505050",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Search Bar can be rendered (without the delete button) matches its previous image 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "marginBottom": 6.67,
      "marginLeft": 0,
    }
  }
>
  <View
    style={
      {
        "borderColor": "red",
        "borderWidth": 0,
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#104e01",
          "borderRadius": 11.25,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 6.67,
          "overflow": "hidden",
          "width": 750,
        }
      }
    >
      <TextInput
        autoCapitalize="none"
        keyboardType="default"
        maxFontSizeMultiplier={2}
        onBlur={[MockFunction]}
        onChangeText={[Function]}
        placeholder="SearchBar placeholder text"
        style={
          {
            "backgroundColor": "white",
            "fontSize": 30.595932917809698,
            "minHeight": 80.03999999999999,
            "paddingLeft": 15,
            "textAlign": "left",
            "textAlignVertical": "center",
            "width": 675,
          }
        }
        testID="searchTermInput"
        value=""
      />
    </View>
  </View>
</View>
`;
